.github/workflows/build-and-deploy.yml

```
@@ -0,0 +1,38 @@
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+      - name: Build with Maven
+        run: mvn install
+
+      - name: Upload war file
+        uses: actions/upload-artifact@v2
```

actionlint

the runner of "actions/upload-artifact@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)